### PR TITLE
Added feature toggling package

### PR DESCRIPTION
**Pull request by JakeGinnivan**

### CODE REVIEW

File: packages/feature-toggles/src/create-feature-updater.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            'Error initialising launch darkly client, falling back to feature file',
+        )
+    }
+
+    if (initialFeatureState) {
+        await writeFeatureFile(config.featureStateFile, initialFeatureState, logger)
+    }
+
+    // We want to read the state from the file to ensure it works.
+    if (!(await featureStateFileExists(config.featureStateFile))) {
```
Comment (antonjb) [inline packages/feature-toggles/src/create-feature-updater.ts]: Personal preference would be for the result of the function call to be a variable outside the if statement

File: packages/feature-toggles/src/launch-darkly/client.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            logger: loggerWithoutReconnErrors,
+            featureStore,
+        })
+
+        const timeout = setTimeout(() => {
+            ldClient.close()
+            reject(
+                new Error(
+                    'Failed to initialise the launch darkly client, ' +
+                        'see https://confluence.swmdigital.io/display/DPT/What+to+do+when+launch+darkly+is+down+when+The+West+starts',
```
Comment (antonjb) [inline packages/feature-toggles/src/launch-darkly/client.ts]: This seems specific to our use case

File: packages/feature-toggles/src/launch-darkly/result-parser.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { StringBooleanMap, StringAnyMap, FeatureState } from '../universal'
+
+export const readLdAllFlagsResult = (features: StringAnyMap) => {
+    const booleanFeatures = Object.keys(features).reduce<StringBooleanMap>((acc, key) => {
+        const toggleValue = features[key]
+
+        // We only support boolean toggles at the moment
+        if (typeof toggleValue === 'boolean') {
+            // Features in launch darkly use - as separators by default
+            // This allows us to continue using _ in code
```
Comment (antonjb) [inline packages/feature-toggles/src/launch-darkly/result-parser.ts]: didn't think we had this requirement any more?

File: packages/feature-toggles/src/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    return feaureStateMiddleware
+
+    function feaureStateMiddleware(
+        req: express.Request & WithFeatures,
+        _res: express.Response,
+        next: express.NextFunction,
+    ) {
+        req.features = featureReceiver.featureState
+
+        if (next) {
```
Comment (antonjb) [inline packages/feature-toggles/src/middleware.ts]: the type doesn't look optional, or is `express.NextFunction` optional in some way?

Comment (JakeGinnivan) [reply packages/feature-toggles/src/middleware.ts]: Agreed!

File: packages/feature-toggles/src/launch-darkly/result-parser.ts (hunk 3 above)
Comment (JakeGinnivan) [reply packages/feature-toggles/src/launch-darkly/result-parser.ts]: Comment isn't correct anymore, removed.

File: packages/feature-toggles/src/launch-darkly/client.ts (hunk 2 above)
Comment (JakeGinnivan) [reply packages/feature-toggles/src/launch-darkly/client.ts]: Moved into readme, nice catch

File: packages/feature-toggles/src/create-feature-updater.ts (hunk 1 above)
Comment (JakeGinnivan) [reply packages/feature-toggles/src/create-feature-updater.ts]: Good call, fixed

**Pull request merged**